src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +589,1762 @@ void M()
+        {
+            // error CS1501: No overload for method 'M' takes 1 arguments
+            var code =
+@"
+<Workspace>
+    <Project Language=""C#"" CommonReferences=""true"" AssemblyName=""A1"">
+        <Document FilePath=""ReferencedDocument"">
+namespace N
+{
+    public class BaseClass
+    {
+        public virtual void M() { }
+    }
+}
+        </Document>
+    </Project>
+    <Project Language=""C#"" CommonReferences=""true""  AssemblyName=""A2"">
+        <ProjectReference>A1</ProjectReference>
+        <Document>
+namespace N
+{
+    public class Derived1: BaseClass
+    {
+        public override void M() { }
+    }
+}
+        </Document>
+    </Project>
+    <Project Language=""C#"" CommonReferences=""true""  AssemblyName=""A3"">
+        <ProjectReference>A1</ProjectReference>
+        <Document>
+namespace N
+{
+    public class Derived2: BaseClass
+    {
+        public override void M() { }
+    }
+}
+        </Document>
+    </Project>
+    <Project Language=""C#"" CommonReferences=""true""  AssemblyName=""A4"">
+        <ProjectReference>A3</ProjectReference>
+        <Document>
+namespace N
+{
+    public class T
+    {
+        public void Test() { 
+            new Derived2().[|M|](1);
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            var fix_All =
+@"
+<Workspace>
+    <Project Language=""C#"" CommonReferences=""true"" AssemblyName=""A1"">
+        <Document FilePath=""ReferencedDocument"">
+namespace N
+{
+    public class BaseClass
+    {
+        public virtual void M(int v) { }
+    }
+}
+        </Document>
+    </Project>
+    <Project Language=""C#"" CommonReferences=""true""  AssemblyName=""A2"">
+        <ProjectReference>A1</ProjectReference>
+        <Document>
+namespace N
+{
+    public class Derived1: BaseClass
+    {
+        public override void M(int v) { }
+    }
+}
+        </Document>
+    </Project>
+    <Project Language=""C#"" CommonReferences=""true""  AssemblyName=""A3"">
+        <ProjectReference>A1</ProjectReference>
+        <Document>
+namespace N
+{
+    public class Derived2: BaseClass
+    {
+        public override void M(int v) { }
+    }
+}
+        </Document>
+    </Project>
+    <Project Language=""C#"" CommonReferences=""true""  AssemblyName=""A4"">
+        <ProjectReference>A3</ProjectReference>
+        <Document>
+namespace N
+{
+    public class T
+    {
+        public void Test() { 
+            new Derived2().M(1);
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            await TestInRegularAndScriptAsync(code, fix_All, index: 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocation_Cascading_OfferFixCascadingForImplicitInterface()
+        {
+            // error CS1501: No overload for method 'M1' takes 1 arguments
```

Comment:
not sure what cascading it does? should it has interface I2 or something? or multiple type that implements I1 for cascading test?

Reply:
*cascading* refers to the *fix_All* case (index:1). The difference is:

* *fix_All* aka *cascading*: fix all **declarations** returned by find all references.
 *fix_DeclarationOnly*: fix only the method declaration of the called method.

The fixer never changes anything besides declarations (call site is never fixed).

Reply:
tagging @jinujoseph @kuhlenh since this behavior is different than signature change even though conceptually what it does is same as what signature change do (change signature of method).

we might want to open a tracking issue to make our existing signature change service to handle this case as well and move this fixer to use that service to have consistent behavior on such code transformation.